hadoop-hdds/rocksdb-checkpoint-differ/src/test/java/org/apache/ozone/rocksdiff/TestRocksDBCheckpointDiffer.java

```diff
@@ -1100,4 +1114,108 @@ public void testPruneOlderSnapshotsWithCompactionHistory(
     deleteDirectory(compactionLogDir);
     deleteDirectory(sstBackUpDir);
   }
+
+  private static Stream<Arguments> sstFilePruningScenarios() {
+    return Stream.of(
+        Arguments.of("Case 1: No compaction.",
+            "",
+            Arrays.asList("000015", "000013", "000011", "000009"),
+            Arrays.asList("000015", "000013", "000011", "000009")
+        ),
+        Arguments.of("Case 2: One level compaction.",
+            "C 000015,000013,000011,000009:000018,000016,000017\n",
+            Arrays.asList("000015", "000013", "000011", "000009", "000018",
+                "000016", "000017", "000026", "000024", "000022", "000020"),
+            Arrays.asList("000015", "000013", "000011", "000009", "000026",
+                "000024", "000022", "000020")
+        ),
```

Comment:
Shouldn't `000018`, `000016` and `000017` remain in the `expectedFiles` list in this case?

Reply:
No actually.

Let's take case 2 for example:
Level 1: "000015", "000013", "000011", "000009"
Level 2: "000018", "000016", "000017", "000026", "000024", "000022", "000020"

files: "000018", "000016", "000017" are from compaction and "000026", "000024", "000022", "000020" are new files at level 2.

For diff, we only need  "000015", "000013", "000011", "000009", "000026", "000024", "000022", "000020" files.
If snapshot is taken before compaction "000015", "000013", "000011", "000009" files are needed to generate diff.
And if compaction is taken after compaction "000015", "000013", "000011", "000009", "000026", "000024", "000022", "000020" are needed. "000018", "000016", "000017" are common files and won't generate any diff.

Reply:
Make sense. I was thinking about the DAG in the other direction.